pkg/controller/integration/integration_controller.go

```diff
@@ -217,6 +218,15 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
+	// Watch ServiceBindings created
+	err = c.Watch(&source.Kind{Type: &sb.ServiceBinding{}}, &handler.EnqueueRequestForOwner{
```

Comment:
I think this may cause issues in case the `ServiceBinding` resource is not present in the cluster (or not yet present when the Camel K operator is installed). We have a similar issue with Knative where we ended up watching the `ReplicaSet` instead of specific Knative resources (right @astefanutti ?). Is there a child standard resource that we can watch?

Reply:
Ok, I see you've already targeting this. There's also a safer way to look if you can watch a resource by using these APIs: https://github.com/apache/camel-k/blob/51a5c234e362aa0815da4e7d8951851dc4b24a39/pkg/util/kubernetes/permission.go#L33

Also adding a watch on sub-resources (if possible) that change can help in case the servicebinding operator is installed after the camel k operator is started.

Reply:
Thanks for reviewing! There is the [status](https://github.com/redhat-developer/service-binding-operator/blob/master/config/crd/bases/operators.coreos.com_servicebindings.yaml#L270) subresource if that helps ?

Reply:
No, I was referring to a child-resource. Like if you know that servicebinding always syncs a secret with some labels when updating the binding resource, then listening for changes in the secrets (all or with filter) triggers the reschedule of the integration, so it proceeds.
Secrets are an example of "watchable" resource being in core v1.

Reply:
That would be useful only in cases where ServiceBinding is installed after camel k.

Reply:
Ah right no not by labels, here's a sample secret:
```
apiVersion: v1
data:
 ...
kind: Secret
metadata:
  name: sample-service-binding-request
  namespace: service-binding
  ownerReferences:
  - apiVersion: operators.coreos.com/v1alpha1
    controller: true
    kind: ServiceBinding
    name: sample-service-binding-request
  selfLink: /api/v1/namespaces/service-binding/secrets/sample-service-binding-request
type: Opaque
```
We could check in the owner references and work our way up to the integration name by checking if it exists but it may be simpler to file a request on the Service Binding Operator side to copy labels from the `ServiceBinding` to the generated `Secret` which seems at first glance like a reasonable request...

Reply:
Tracked with https://github.com/redhat-developer/service-binding-operator/issues/852